src/Stats.ts

```diff
@@ -1,96 +1,83 @@
 import {Rune} from './Rune'
+import {Stat} from './Stat'
+import {Character} from './Character'
 
 export class Stats {
-  minDexterity: number = 35
-  _dexterity: number = 40
-  maxDexterity: number = 100
+  private readonly _dexterity: StatInternal
```

Comment:
We declare our private stat instances as `StatInternal` which we define later, so that we can access setters for `value` (which no class other than `Stats` should reasonably use as designed) and have getters that return `Stat`.